feat: redesign website

anhek opened the pull request:
Hello :)

I have finished the redesign. There's still some things to do like : 

- adding the colours on the tags
- customizing the form

I did some research and it looks like you need to have a paid subscription on Upscribe to have the ability to add custom theme :/

I'd like someone to help me getting the cards to have the same height like on this article please : https://www.lottejackson.com/learning/an-equal-height-grid-using-flexbox

<img width="750" alt="equal-height-grid" src="https://user-images.githubusercontent.com/1397483/67051222-d4142d00-f13a-11e9-9046-4ea0f61c5b42.png">

Feedback time ! 🙂

swapagarwal commented on src/pug/includes/body.pug:
```diff
@@ -3,24 +3,30 @@ include ./fork.pug
+					img(src="/assets/img/devswag-logo.svg", alt="DevSwag")
+					span.alt-logo DevSwag
+				h2 swag opportunities for developers 😎
+				p As a developer, nothing piques my interest more than developer swag!
```
Let's make this text bolder for more contrast?

swapagarwal commented:
The bottom subscribe form's background makes the text harder to read. Let's change that.

swapagarwal commented:
I love the new prominent check it out button! ❤️

anhek commented:
> * The bottom subscribe form's background makes the text harder to read. Let's change that.
> * I love the new prominent check it out button! ❤️
> * It just hit me that we should keep all the filtering options (difficulty, tags, expired) together and move the sort option to the right. Thoughts?

I made some updates for the newsletter background, the heading font-sizes which is now bigger and I added the tags colours.
For the filtering options, It doesn't bother me to have a nav bar like, it gives more spaces for the content.

aslafy-z commented on package.json:
```diff
@@ -13,7 +13,9 @@
   },
-  "dependencies": {},
+  "dependencies": {
+    "acorn": "^7.1.0"
```
Can you please drop that change?

aslafy-z commented on package.json:
```diff
@@ -23,7 +25,7 @@
     "eclint": "2.8.1",
     "got": "9.6.0",
-    "gulp": "4.0.2",
+    "gulp": "^4.0.2",
```
Please revert that change, we use Renovate to manage and update our dependencies.

anhek replied:
Sorry for the updates, I’ll do it asap

aslafy-z commented on src/styl/imports/card.styl:
```diff
@@ -1,67 +1,107 @@
 
 	.img-container
-		height 300px
+		height 260px
```
We might need to adjust image size in the build process to match this.

aslafy-z commented on src/styl/imports/colors.styl:
```diff
@@ -0,0 +1,26 @@
+link-color = #67d0a1
+link-hover-color = lightness(link-color, 80%)
+
+// TAGS
```
Is there a way we could automatically generate colors for these tags instead of hardcoding it?

anhek replied:
Do you have any ideas about how to implement this ? I'm not sure

aslafy-z replied:
I love it! Well done!

aslafy-z commented on src/styl/imports/universal.styl:
```diff
@@ -3,6 +3,7 @@
 	margin 0
 	box-sizing border-box
 	text-rendering optimizeLegibility
+	// border 1px solid red
```
Please drop that comment

aslafy-z commented:
Can we find some svg to replace `tiny_grid.png`?

anhek commented:
Why the Travis CI build is failing on " 1) Google Assistant has a valid reference" ? 🤔

vikaspotluri123 commented:
Don't worry too much about CI, it's mostly just to make sure opportunities are still functional 😄

aslafy-z commented on package.json:
```diff
@@ -17,7 +17,7 @@
     "@babel/core": "7.6.4",
     "@babel/preset-env": "7.6.3",
-    "autoprefixer": "9.7.0",
+    "autoprefixer": "9.6.5",
```
Can you please revert that change?

aslafy-z replied:
And reset the package-lock.json file also.

aslafy-z commented on data.json:
```diff
@@ -98,15 +98,6 @@
     "dateAdded": "2018-12-03T04:36:19.000Z",
     "tags": ["clothing", "expired"]
   },
-  {
```
Please reset that file.

aslafy-z commented:
Please drop the png versions of the tiny_grid file. I guess we can also put the svg one to a own file?

anhek commented:
> Please drop the png versions of the tiny_grid file. I guess we can also put the svg one to a own file?

It's done. Is it best to leave the svg code inside the index.styl or having the code inside a svg file in the img folder ?

aslafy-z commented:
I prefer keeping it in a separate file, we can do inlining at build time if we want

swapagarwal commented:
@anhek Ping.

anhek commented:
Sorry for the late reply I was off, I'll take a look :)

aslafy-z commented:
I'll rebase this PR for you @anhek!
Please re-clone the repo after I made the change ;)

anhek commented:
> I'll rebase this PR for you @anhek!
> Please re-clone the repo after I made the change ;)

Thank you ! I I'll do that :)

aslafy-z commented on src/styl/index.styl:
```diff
@@ -8,38 +27,63 @@
+		color white
+
+	.hero
+		background url('/assets/img/tiny_grid_@2X.png') repeat
```
Why is this still here? Did I broke something?

aslafy-z commented:
Should be good by now, but it looks like some changes are missing, it's weird. Can you check?

anhek commented:
I updated the tiny_grid background as a svg file, as you were previously asking. I hope it's all good now. 

But there's a missing link in the data.json, the image for the CircleCI "https://hacktoberfest.circleci.com/img/orbtoberfest-header_graphic.4f2bfde4.svg" is not available anymore which makes the gulp process fail.

aslafy-z commented:
Just fixed that dead link in master and merged it back to your branch. It builds!

swapagarwal commented:
@aslafy-z @vikaspotluri123 Let's get this merged? 🤠

aslafy-z commented on src/pug/includes/body.pug:
```diff
@@ -3,24 +3,30 @@ include ./fork.pug
+					span.alt-logo DevSwag
+				h2 swag opportunities for developers 😎
+				p As a developer, nothing piques my interest more than developer swag!
+					br
```
Not sure a our that alignment, can you check?

swapagarwal replied:
@anhek

aslafy-z commented:
A last point about the advertisment: see this screenshot

![Screenshot_20191119_122013_com android chrome](https://user-images.githubusercontent.com/8191198/69142481-314b2780-0ac7-11ea-8236-69da0d90ac66.jpg)

swapagarwal commented:
The feedback and announcement divs are also overlaying on the content a bit, so I guess the GitHub ribbon is okay.

anhek commented:
Sorry I was a bit busy working on my portfolio these past few weeks. I'll take a look on the design to see if there's some small improvements that can be made.

anhek commented:
Hello everyone 😃 

I'm proposing to move the Github button which was in the top right corner to the end of the hero banner. This way, it doesn't parasite the display on mobile version and we have a clear button to redirect people on the GitHub repo. What do you think about this update ?

I also updated the "Contact Us" button style it doesn't need to be this visible and I also deleted the underline decoration on links.

![Capture d’écran - 2019-12-09 à 12 51 51](https://user-images.githubusercontent.com/1397483/70433733-b8663c80-1a82-11ea-9466-63a17e4dccbc.png)

swapagarwal commented:
This looks good.
@vikaspotluri123 @aslafy-z Do you have any other suggestions?

anhek commented:
Oh and by the way, I also added the applause implementation from @pannapat in my master branch 🙂

swapagarwal commented:
Pinging @aslafy-z and @vikaspotluri123 😅

aslafy-z commented:
I like this GH button! Please implement it.

I'm not sure whether we should merge as-is, it becomes eavyloaded with the contact ribbon, feedback and announcement buttons and the ads.

 ![](https://i.imgur.com/2mBxbJh.jpg)

anhek commented:
You're right, the website is overloaded with too much stuff in overlay 🤔

swapagarwal commented:
So the GitHub ribbon will be removed.
Let's hide the div containing ads for now.
Anything other suggestions?

vikaspotluri123 commented:
Maybe put feedback and github links in the footer? 🤔

swapagarwal commented:
Yep, let's do that. 👍

sampoder commented:
Wow! This looks amazing! going randomly drop in but i was thinking the logo has great colours but looks a little weird how it's peeling, I'm not sure but i dont think it looks right. Also, ik it's a bit cluttered but maybe a dark mode..... just maybe. Still awesome though

SemperDisco commented:
This looks great! Is anyone still working on it? I think it would be great if the conflict is fixed and we can ship this! :)

swapagarwal commented:
@anhek @aslafy-z Let's ship this? 🙃

vikaspotluri123 commented:
@swapagarwal I thought we wanted to put feedback / GH links in the footer?

swapagarwal commented:
@vikaspotluri123 Yes.
@anhek Can you incorporate Vikas' feedback? 🙂

SemperDisco commented:
If @anhek isn't active anymore, I'm also willing to complete this PR.

vikaspotluri123 commented:
@anhek thanks! Can you rebase against master? I've fixed the issue 😄

SemperDisco commented:
The build and deploy failed probably because a connection error during the downloading of the images. The same happened in my PR. Running the build and deploy again fixed it.  

@swapagarwal, @aslafy-z, or @vikaspotluri123 can you give this a try?

anhek commented:
> @anhek thanks! Can you rebase against master? I've fixed the issue 😄

When I'm rebasing using the command : `git rebase upstream/master`, I get an older version of the website and the error is this present. It is normal or am I doing something wrong ?

vikaspotluri123 commented:
@SemperDisco the error was

> 404 returned from `https://www.datarequests.org/img/blog/hacktoberfest-2019.png`

I've fixed it by using a different image

@anhek sounds like you got it figured out, let me know if you have any other questions 👍

anhek commented:
> @SemperDisco the error was
> 
> > 404 returned from `https://www.datarequests.org/img/blog/hacktoberfest-2019.png`
> 
> I've fixed it by using a different image
> 
> @anhek sounds like you got it figured out, let me know if you have any other questions 👍

Alright, I made a couple of updates. I want to `npm run-script lint:fix` before committing but I get the error :

<img width="764" alt="Capture d’écran 2020-04-16 à 17 11 57" src="https://user-images.githubusercontent.com/1397483/79473432-68395100-8005-11ea-87bd-c3fef18d2805.png">

```
function handleSort() {
	Array.from(contentElement.children)
		.map(child => contentElement.removeChild(child)) // eslint-disable-line unicorn/prefer-node-remove
		.sort(sort[sortingInput.value])
		.forEach(sortedChild => contentElement.append(sortedChild));
}
```

What should I do ?

vikaspotluri123 commented:
Have you run `npm install` after rebasing? I think the update to xo might put your ruleset out of sync

anhek commented:
> Have you run `npm install` after rebasing? I think the update to xo might put your ruleset out of sync

I totally forgot in the first place but I just did a `npm install`. 
If I put the comment line back `// eslint-disable-line unicorn/prefer-node-remove` on the `index.js` file, will the Travis CI test be good ?

vikaspotluri123 commented:
I think so 😄 the only test that's failing is the disable that was removed

anhek commented:
I'll make a new commit and it should be good then ^^

Also you can check the updates I did for the Github buttons on the header and the footer. 
For the feedback link in the footer, I'm not sure about what you want to do. Maybe someone can take care of it ? 🙂

SemperDisco commented:
Great job @anhek!

anhek commented:
Thank you guys 😊🙌🎉

aslafy-z commented on .gitignore:
```diff
@@ -79,5 +79,5 @@ typings/
 .serverless
 
-
+.DS_Store
```
```suggestion
# End of https://www.gitignore.io/api/node

.DS_Store
```

aslafy-z replied:
Please move this entry below the `# End of https://**`

aslafy-z commented on package-lock.json:
```diff
@@ -2583,7 +2583,7 @@
     },
```
Please revert changes to this file

aslafy-z commented on src/pug/includes/body.pug:
```diff
@@ -45,38 +56,57 @@ include ./fork.pug
 				- difficulty = swag.difficulty + ' difficulty';
 				- difficultyTitle = 'Difficulty: ' + swag.difficulty;
 				- expired = swag.tags.includes('expired');
+				- reference = 'https://devswag.io' + swag.reference ? swag.reference : swag.name;
```
I can't understand why the applause-button `url` argument are not prefixed by `https://devswag.io`...
Anyway, something like this would be better if we stick with the prefix:
```suggestion
				- reference = 'https://devswag.io/#' + swag.reference ? swag.reference : swag.name;
```

aslafy-z commented on src/pug/includes/head.pug:
```diff
@@ -12,6 +12,7 @@ link(rel="apple-touch-icon" size="128x128" href="/assets/img/logo.png")
 
 link(rel="stylesheet" href=bustedAssets.css)
 link(rel='stylesheet' href='https://cdn.jsdelivr.net/npm/mobius1-selectr@2.4.8/dist/selectr.min.css')
+link(rel='stylesheet' href='https://unpkg.com/applause-button/dist/applause-button.css')
```
```suggestion
link(rel='stylesheet' href='https://cdn.jsdelivr.net/npm/applause-button@3.3.1/dist/applause-button.css')
```

aslafy-z commented on src/pug/index.pug:
```diff
@@ -12,3 +12,4 @@ html(lang='en')
 
 		script(defer src=bustedAssets.js)
 		script(defer src='https://cdn.jsdelivr.net/npm/mobius1-selectr@2.4.8/dist/selectr.min.js')
+		script(defer src='https://unpkg.com/applause-button/dist/applause-button.js')
```
```suggestion
		script(defer src='https://cdn.jsdelivr.net/npm/applause-button@3.3.1/dist/applause-button.js')
```

aslafy-z commented:
I added a few quick points.. I can still see the contact ribbon being broken from my side (https://github.com/swapagarwal/swag-for-dev/pull/443#issuecomment-572156639), it could eventually be part of another PR. WDYT?

Anyway, thank you @anhek for your hard work!

aslafy-z commented:
Will apply my comments myself so we can ship ASAP!

aslafy-z commented:
I've just seen the contact ribbon was moved to the bottom, should we drop the one that sticks to the top?

swapagarwal commented:
@aslafy-z What's the mobile bug?

aslafy-z commented:
@swapagarwal see below:

![Screenshot_20200419_225736_com android chrome~2](https://user-images.githubusercontent.com/8191198/79699713-6e436200-8291-11ea-85db-8174849abe3f.jpg)

swapagarwal commented:
Ah, okay! I was browsing from Firefox Focus earlier.
It's reproducible on Chrome. Let's remove that.

swapagarwal commented:
@anhek

aslafy-z commented:
I went ahead and applied the following:
- Remove applause buttons
- Restore .gitignore
- Delete old fork.pug
- Delete the top banner

Are we ready to go? @swapagarwal @vikaspotluri123?

swapagarwal commented:
Looks good! 🚀

vikaspotluri123 commented:
🚀 ✨ 🎉

swapagarwal commented:
Thanks everyone! 🔥
The new site looks great!! 🔥🔥
